src/main/java/eu/europeana/api/recommend/util/RecommendServiceUtils.java

```diff
@@ -0,0 +1,170 @@
+     * @param jsonObject
+     * @return
+     */
+    public List<String> extractItems(JSONObject jsonObject) {
+        List<String> items = new ArrayList<>();
+        try {
+            JSONArray setsList = jsonObject.getJSONArray(ITEMS_FIELD);
+            for(int i =0; i < setsList.length(); i++) {
+                JSONObject set = (JSONObject) setsList.get(i);
+                JSONArray itemList =  set.getJSONArray(ITEMS_FIELD);
+                for(int j =0; j < itemList.length(); j++) {
+                 items.add(StringUtils.substringAfter((String) itemList.get(j), "http://data.europeana.eu/item"));
+                }
+            }
+        } catch (JSONException e) {
+            LOG.error("Error fetching the objects from set api response. {}", e.getMessage());
+        }
+        return items;
+    }
+
+    /**
+     * Extracts skos:prefLabel(s) and skos:altLabel(s)
+     *
+     * @param jsonObject
+     * @return
+     */
+    public List<String> extractLabels(JSONObject jsonObject) {
+        List<String> labels = new ArrayList<>();
+        try {
+            JSONArray items =  jsonObject.getJSONArray(ITEMS_FIELD);
+            for(int i =0; i < items.length(); i++) {
+                JSONObject item = (JSONObject) items.get(i);
+                JSONObject prefLabel = item.getJSONObject(PREF_LABEL_FIELD);
+                JSONObject altLabel = item.getJSONObject(ALT_LABEL_FIELD);
+                labels.addAll(getPreflabels(prefLabel.toString()));
+                labels.addAll(getAltLabels(altLabel.toString()));
+            }
+        } catch (JSONException e) {
+            LOG.error("Error fetching the objects from entity api response. {}", e.getMessage());
+        }
+        return labels;
+    }
+
+    /**
+     * Returns all the prefLabel values of all languages
+     * @param prefLabels
+     * @return
+     */
+    private Collection<String> getPreflabels(String prefLabels) {
+       try {
+           ObjectMapper mapper = new ObjectMapper();
+           return mapper.readValue(prefLabels, new TypeReference<HashMap>(){}).values();
+        } catch (JsonProcessingException e) {
+           LOG.error("Error reading the prefLabels. {}", e.getMessage());
+        }
+        return new ArrayList();
+    }
+
+    /**
+     * Returns all the altLabel values of all languages
+     * @param altLabels
+     * @return
+     */
+    private List<String> getAltLabels(String altLabels) {
+       Map<String, List<String>> map ;
+       List<String> altLabel = new ArrayList<>();
+       try {
+           ObjectMapper mapper = new ObjectMapper();
+           map = mapper.readValue(altLabels, new TypeReference<HashMap>(){});
+           map.entrySet().stream().forEach((entry) -> {
+                for(String value : entry.getValue()) {
+                    altLabel.add(value);
+                }});
+        } catch (JsonProcessingException e) {
+           LOG.error("Error reading the altLabels. {}", e.getMessage());
```

Comment:
SQ issue: "Log or rethrow this exception"

Reply:
logged exception